File: 2022/day_1/rust/src/main.rs
fn main() {
    let contents: &str = include_str!("../../input.txt");
    let elves: Result<Vec<Vec<u64>>,_>  = contents
        .trim()
        .split("\n\n")
        .map(|lines| lines.lines().map(|s| s.parse()).collect())
        .collect();
    let elves = elves.unwrap();
    let mut calories: Vec<u64> = elves.iter().map(|nums| nums.iter().sum()).collect();

    calories.sort();
    
    // Part 1
    println!("Calories: {:?}", calories.last().unwrap());

    // Part 2
    println!("Part 2: {:?}", calories.iter().rev().take(3).sum::<u64>());
}

File: 2022/day_1/main.go
package main

import(
    "fmt"
    "io/ioutil"
    "strings"
    "strconv"
    "sort"
)


func main() {
    // Open file
    content, error := ioutil.ReadFile("input.txt")
    if error != nil {
        panic(error)    
    }

    elves := strings.Split(string(content), "\n\n")
    var calories []int
    for _, elf := range elves { 
        caloriesInElf := strings.Split(elf, "\n")
        totalCalories := 0
        for _, calorieString := range caloriesInElf {
            calorieInt, _ := strconv.Atoi(calorieString)
            totalCalories += calorieInt
        }
        calories = append(calories, totalCalories)
    }
    sort.Ints(calories)
    
    // Part 1
    fmt.Println(calories[len(calories) - 1])

    // Part 2
    fmt.Println(calories[len(calories) - 1] + calories[len(calories) - 2] + calories[len(calories) - 3])
}
